Error message in server logs on CSV export error

Facyla opened the pull request:
Provides a useful feedback message when CSV export fails due to wrong permissions on wp-content/upload/ folder (or any other folder set by WC CSV Exporter module). 

This helps understand why CSV export fails under some conditions, by providing a hint on the error cause, instead of silently failing.

### Testing instructions

Code review is probably sufficient; manual testing is a brittle process however you can certainly try the following steps. Start by adding this snippet to a suitable location, such as `wp-content/mu-plugins/pr-34802.php` (you can and should remove it afterwards).

```php
<?php
/**
 * Attempts to sabotage the CSV export process by modifying the upload directory to a 
 * path that is read-only.
 */
add_action( 'wp_ajax_woocommerce_do_ajax_product_export', function () {
	$test_parent_dir = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'test_34802_';
	$test_child_dir  = $test_parent_dir . DIRECTORY_SEPARATOR . 'uploads';

	if ( ! file_exists( $test_parent_dir ) ) {
		mkdir( $test_parent_dir, 0444 );
	}

	add_filter( 'upload_dir', function ( array $upload_dir ) use ( $test_child_dir ): array {
		$upload_dir['basedir'] = $test_child_dir;
		return $upload_dir;
	} );
}, 5 );
```

With that in place:

1. Make sure you have some test products (you don't need many).
2. Navigate to **Products ▸ All Products ▸ Export** and export your products by clicking on **Generate CSV**.
3. If the snippet worked, this should still result in a CSV file being generated, but it will only contain the first row (the column headings).
4. Now navigate to the error logs found in **WooCommerce ▸ Status ▸ Logs** and you should find an entry looking like the following (the appearance of the error logs may be different in your case, that's ok—but it should be possible to identify an entry reading *"Unable to create or write to &lt;FILENAME&gt; during CSV export"):*

<img width="981" alt="abstract-csv-export-error" src="https://user-images.githubusercontent.com/3594411/213521367-0e256ef9-aeb8-421a-8bfc-82de0fa9f681.png">

Facyla committed:
Error message in logs on CSV export error

Provide a useful feedback message when CSV export fails due to wrong permissions on wp-content/upload/ folder (or any other folder set by WC CSV Exporter module). 

This helps understand why CSV export fails under some conditions, by providing a hint on the error cause, instead of silently failing.

Facyla committed:
Merge pull request #1 from Facyla/Facyla-patch-error-msg-csv-export

Error message in logs on CSV export error

Facyla commented:
Hi, 
I see that this issue is blocked: how can i help?  I may provide another PR or update this one if it doesn't meet some requirement.

barryhughes commented on plugins/woocommerce/includes/export/abstract-wc-csv-batch-exporter.php:
```diff
@@ -127,6 +127,7 @@ public function generate_file() {
 	protected function write_csv_data( $data ) {
 
 		if ( ! file_exists( $this->get_file_path() ) || ! is_writeable( $this->get_file_path() ) ) {
+			 error_log(__("ERROR : Cannot create temporary CSV export file : please check permissions on upload directory.", 'woocommerce'));
```
Instead of the system error log, it may be better to use the WooCommerce error log. We can also change the quotation style, and fix a small linting issue (whitespace):

```suggestion
			wc_get_logger()->error( 
				sprintf(
					/* translators: %s is file path. */
					__( 'Unable to create or write to %s during CSV export. Please check file permissions.', 'woocommerce' ),
					esc_html( $this->get_file_path() )
				)
			);
```

Facyla committed:
Update plugins/woocommerce/includes/export/abstract-wc-csv-batch-exporter.php

use the WooCommerce error log, change the quotation style, and fix a small linting issue (whitespace)

Co-authored-by: Barry Hughes <3594411+barryhughes@users.noreply.github.com>

barryhughes committed:
Improve error logging when file permissions prevent CSV generation.

barryhughes reviewed (dismissed):
LGTM, thanks for the contribution :+1:

- Changelog added
- Testing instructions added
- Will merge once checks pass

barryhughes committed:
Linting (clean-up whitespace).
